### PR TITLE
Add utility to wait until the tun is up and running

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1204632863750325/f

### Description
See asana task

### Steps to test this PR
NA

### CODE REVIEW

File: src/netguard/tun.c [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+/**
+ * Returns whether the given address represents a public IP address or not
+ * @param address IPv4 or IPv6 address
+ * @return `true` if it's public IP address, false otherwise
+ */
+static bool is_public_ip(const struct sockaddr_storage* address);
+
+/**
+ * Sends random data to a random IP public IP address
```
Comment (nshuba) [inline src/netguard/tun.c:32]: Nit: typo "Sends random data to a random IP public IP address" --> "Sends random data to a random public IP address." And below: "syccess" --> "success"

File: src/netguard/tun.c [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            return -1; // Timeout occurred
+        }
+
+        int ready = select(tun_fd + 1, &fd_set, NULL, NULL, &timeout);
+        if (ready > 0) {
+            return 0; // Tunnel device is ready
+        }
+
+        FD_SET(tun_fd, &fd_set);
+        try_sending_random_udp(is_ipv6_enabled);
```
Comment (nshuba) [inline src/netguard/tun.c:70]: Do we care if this succeeded or failed?

Comment (aitorvs) [reply src/netguard/tun.c:70]: Not really. This is just to ensure there's traffic to be routed through the tunnel. The important part is detecting if the `tun` is ready to read in the `select`

**Comment by aitorvs** [comment]
@nshuba I remove IPv6 from the equation. It was so it we failed sending a random packet on IPv6 we'd anyway try IPv4. So it was making code more difficult to understand and effectively not giving a lot of value.

LMK what you think

File: src/netguard/include/tun.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+#define NETGUARD_TUN_H
+
+/**
+ * Waits  for the tunnel device to become ready for reading by periodically checking the file descriptor using
+ * `select`.
+ * It also sends a random UDP packet in each iteration to ensure traffic is being routed through the tunnel.
+ * If the tunnel becomes ready, it returns success, else failure
+ *
+ * @param tun_fd tunnel file descriptor
+ * @param is_ipv6_enabled whether IPv6 is enabled or not
```
Comment (karlenDimla) [inline src/netguard/include/tun.h:11]: This could be removed.

**Review by karlenDimla — APPROVED** [review]
Small NITs, but from how I understand the code I think it looks good.

**Pull request merged**